MapboxNavigation/NavigationMapView.swift

```diff
@@ -342,9 +343,7 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
             let function: CAMediaTimingFunction? = animated ? CAMediaTimingFunction(name: kCAMediaTimingFunctionLinear) : nil
             setCamera(newCamera, withDuration: duration, animationTimingFunction: function, edgePadding: padding, completionHandler: nil)
         } else {
-            UIView.animate(withDuration: duration, delay: 0, options: [.curveLinear, .beginFromCurrentState], animations: {
-                self.userCourseView?.center = self.convert(location.coordinate, toPointTo: self)
-            }, completion: nil)
+            self.userCourseView?.center = self.convert(location.coordinate, toPointTo: self)
```

Comment:
This method gets called on every location update. So removing the `UIView.animate(withDuration:delay:options:animation:)` block means the course view never animates, even between user location updates. `duration` is either 1 or 0 depending on whether the caller wants animation.

In #1505, I meant that we should _also_ call `UIView.animate(withDuration:delay:options:animation:)` with a `duration` of 0 elsewhere, to temporarily disable the implicit animation that happens whenever the view’s `center` changes.

Reply:
@1ec5 fixed.